aggregator/internal/pkg/aggregator.go

```diff
@@ -178,6 +178,8 @@ func NewAggregator(aggregatorConfig config.AggregatorConfig) (*Aggregator, error
 		telemetry:             aggregatorTelemetry,
 	}
 
+	go aggregator.clearTasksFromMaps(garbageCollectorPeriod, garbageCollectorTasksAge)
```

Comment:
This doesn't look like the right place to instantiate the task. I would do it at the call-site. Here we're just asking for a struct to be initialized and returned.

Reply:
done